### PR TITLE
Fix possible division by zero in TwoHandMoveLogic

### PR DESCRIPTION
**Pull request by MenelvagorMilsom**
## Overview
For positionless controllers like xbox controller, the pointer position and the head position were coincident at the head position. This change ensures that if this is the case, we do not perform a division by zero in `TwoHandMoveLogic`, as this was causing objects to disappear into infinity.

## Changes
- Fixes: #5829.

### CODE REVIEW

File: Assets/MixedRealityToolkit.Tests/PlayModeTests/ManipulationHandlerTests.cs [hunk 1]
```diff
@@ -856,6 +856,59 @@ public IEnumerator ManipulationHandlerRotateHeadGGV()
...
+            Vector3 correction = hand.GetPointer<GGVPointer>().Position - cameraPosition;
+            yield return hand.Move(-correction, numHandSteps);
+            yield return null;
+
+            Assert.AreEqual(expectedDist, Vector3.Distance(testObject.transform.position, cameraPosition), 0.01f);
+
+            yield return hand.SetGesture(ArticulatedHandPose.GestureId.Pinch);
+            yield return null;
+
+            Assert.AreEqual(expectedDist, Vector3.Distance(testObject.transform.position, cameraPosition), 0.01f);
```
Comment (thalbern) [inline Assets/MixedRealityToolkit.Tests/PlayModeTests/ManipulationHandlerTests.cs]: can you add a movement and verify the position is still valid after grabbing it ? and maybe another verification after you've released it again

**Comment by keveleigh** [comment]
image

Is this meant to be going into that feature branch?

File: Assets/MixedRealityToolkit/Utilities/Physics/TwoHandMoveLogic.cs [hunk 2]
```diff
@@ -35,6 +37,10 @@ public void Setup(MixedRealityPose pointerCentroidPose, Vector3 grabCentroid, Mi
         {
             Vector3 headPosition = CameraCache.Main.transform.position;            
             pointerRefDistance = Vector3.Distance(pointerCentroidPose.Position, headPosition);
+            if (pointerRefDistance == 0)
```
Comment (Troy-Ferrell) [inline Assets/MixedRealityToolkit/Utilities/Physics/TwoHandMoveLogic.cs]: nit: simplify to 
`pointerPosIndependentOfHead = pointerRefDistance != 0;`

**Comment by MenelvagorMilsom** [comment]
> Is this meant to be going into that feature branch?

This branch is where I was planning on making most ManipulationHandler changes, but this can go directly into mrtk_development either. Makes no odds to me.

**Pull request merged**